src/Protocols/Wishbone/Hedgehog.hs

```diff
@@ -0,0 +1,294 @@
+      -> Signal dom (WishboneS2M a)
+      -> ( Signal dom (WishboneS2M a)
+         , Signal dom (WishboneM2S addressWidth (BitSize a `DivRU` 8) a))
+
+    go [] lastRep (_:-m2s) ~(_:-s2m) =
+        B.bimap (emptyWishboneS2M :-) (emptyWishboneM2S :-) $ go [] lastRep m2s s2m
+
+    go (st:stalls) lastRep (m:-m2s) ~(_:-s2m)
+      -- not in a bus cycle, just pass through
+      | not (busCycle m) =
+          B.bimap (emptyWishboneS2M:-) (emptyWishboneM2S:-)
+            (go (st:stalls) lastRep m2s s2m)
+    go (st:stalls) Nothing (m:-m2s) ~(s:-s2m)
+      -- received a reply but still need to stall
+      | busCycle m && strobe m && st > 0 && hasTerminateFlag s =
+          B.bimap
+            -- tell the master that the slave has no reply yet
+            (emptyWishboneS2M :-)
+            -- tell the slave that the cycle is over
+            (emptyWishboneM2S :- )
+            (go (st - 1:stalls) (Just s) m2s s2m)
+
+      -- received a reply but still need to stall
+      | busCycle m && strobe m && st > 0 && not (hasTerminateFlag s) =
+          B.bimap
+            -- tell the master that the slave has no reply yet
+            (emptyWishboneS2M :-)
+            -- tell the slave that the cycle is over
+            (m :- )
+            (go (st - 1:stalls) Nothing m2s s2m)
+
+      -- done stalling, got a reply last second, pass through
+      | busCycle m && strobe m && st == 0 && hasTerminateFlag s =
+          B.bimap
+            (s :-)
+            (m :-)
+            (go stalls Nothing m2s s2m)
+      -- done stalling but no termination signal yet, just pass through to give the slave
+      -- the chance to reply
+      | busCycle m && strobe m && st == 0 && not (hasTerminateFlag s) =
+          B.bimap
+            (emptyWishboneS2M :-)
+            (m :-)
+            (go (0:stalls) Nothing m2s s2m)
+      -- master cancelled cycle
+      | otherwise = B.bimap (emptyWishboneS2M :-) (m :-) (go stalls Nothing m2s s2m)
+    go (st:stalls) (Just rep) (m:-m2s) ~(_:-s2m)
+      -- need to keep stalling, already got the reply
+      | busCycle m && strobe m && st > 0 =
+          B.bimap
+            -- keep stalling
+            (emptyWishboneS2M :-)
+            -- tell the slave that the cycle is over
+            (emptyWishboneM2S :-)
+            (go (st - 1:stalls) (Just rep) m2s s2m)
+      -- done stalling, give reply
+      | busCycle m && strobe m && st == 0 =
+          B.bimap
+            (rep :-)
+            (emptyWishboneM2S :-)
+            (go stalls Nothing m2s s2m)
+      -- master cancelled cycle
+      | otherwise = B.bimap (emptyWishboneS2M :-) (m :-) (go stalls Nothing m2s s2m)
+
+-- | Create a wishbone 'Standard' circuit to drive other circuits.
+driveStandard
+  :: forall dom a addressWidth
+  . ( C.ShowX a, Show a, C.NFDataX a, C.KnownNat addressWidth, C.KnownDomain dom
+    , C.KnownNat (C.BitSize a) )
+  => ExpectOptions
+  -> [WishboneMasterRequest addressWidth a] -- ^ Requests to send out
```

Comment:
```suggestion
  -> [WishboneMasterRequest addressWidth a] -- ^ Requests to sent out
```

Reply:
isn't "send" correct here? "sent" would be past-tense, no?

Reply:
whoops yes.